### DISCUSSION TITLE
Disabling configuration working for one property but not the other

### DISCUSSION THREAD

**Discussion by timja** [renovatebot/renovate#30555] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Hi

I'm trying to migrate Jenkins from dependabot to renovate fully.
see https://github.com/jenkinsci/jenkins/pull/9459

I've managed all but one dependency (with some help from @viceice ❤️ )

We have a niche case, where we do the following:
1. bundle the latest version of `org.jenkins-ci.main:remoting`
2. test against an old version of it
3. have a minimum supported version that people can use

1. should be updated and is working
2. should not be updated and is working with https://github.com/timja-org/jenkins/blob/489c98fe32a59a8fba8ea12bbf5a7ebf710a5b45/.github/renovate.json#L69-L73
3. should not be updated but currently is in https://github.com/timja-org/jenkins/pull/9

I would have expected what was done in 2. to skip it, any ideas?

### Logs (if relevant)

I couldn't find any that were terribly relevant but some snippets + the full log at: https://gist.github.com/timja/6ef849b99cd8f1e093608fbbc3d01368

<details><summary>Logs</summary>

```

          {
            "currentValue": "3107.v665000b_51092",
            "currentVersion": "3107.v665000b_51092",
            "currentVersionTimestamp": "2024-07-26T05:16:14.000Z",
            "datasource": "maven",
            "depName": "org.jenkins-ci.main:remoting",
            "editFile": "pom.xml",
            "fileReplacePosition": 3753,
            "fixedVersion": "3107.v665000b_51092",
            "groupName": "remoting.minimum.supported.version",
            "isSingleVersion": true,
            "packageName": "org.jenkins-ci.main:remoting",
            "packageScope": "org.jenkins-ci.main",
            "registryUrl": "https://repo.jenkins-ci.org/public",
            "registryUrls": [
              "https://repo.jenkins-ci.org/public/",
              "https://repo.maven.apache.org/maven2"
            ],
            "sourceUrl": "https://github.com/jenkinsci/remoting",
            "versioning": "maven",
            "warnings": [],
            "updates": [
              {
                "bucket": "major",
                "newVersion": "3256.v88a_f6e922152",
                "newValue": "3256.v88a_f6e922152",
                "releaseTimestamp": "2024-06-18T23:40:05.000Z",
                "newMajor": 3256,
                "newMinor": 0,
                "newPatch": 0,
                "updateType": "major",
                "branchName": "renovate/major-remoting.minimum.supported.version"
              }
            ]
          },

```

</details>

**Comment by viceice** [comment]
I'll try to look at it after lunch

**Comment by secustor** [comment]
Your rule the fileMatch rule is pointing to `test/pom.xml`, but the file updated is `pom.xml`. 
Currently your configuration basically says: 
```
filename IN [ 'test/pom.xml' ]
AND
packageName IN [ 'org.jenkins-ci.main:remoting' ]
```

If you want to match that rule you should change `matchFileNames` in that case.

**Reply by timja** [reply]
But case 1 is also declared in that file as remoting.version.

I was expecting disabling updates in test/pom.xml would mean there’s nothing to update so it wouldn’t update it

Thoughts?

**Reply by timja** [reply, marked as answer]
I think this will workaround it: https://github.com/timja-org/jenkins/commit/0cf50d27255e2c7ffb92ca9604ee1195fda12494

suggested in https://stackoverflow.com/a/76397580

another option is an external properties file like in https://sergiuoltean.com/2017/03/27/maven-share-properties-between-modules/

These are both workarounds and it would be good to be able to disable just property in some way though

**Reply by timja** [reply]
I'll go with https://github.com/renovatebot/renovate/discussions/30555#discussioncomment-10222150 unless there's any better ideas, thanks!
